orttraining/orttraining/python/training/ortmodule/_utils.py

```diff
@@ -99,6 +99,18 @@ def get_device_from_module(module):
         pass
     return device
 
+
+def get_device_from_inputs(args, kwargs):
+    '''Returns device from first PyTorch Tensor within args or kwargs'''
+
+    device = None
+    if args:
+        device = torch.device(args[0].device)
+    elif kwargs:
+        device = torch.device(next(iter(kwargs.values())).device)
+    return device
```

Comment:
args and kwargs can be None, raising an exception

is it guaranteed that args and kwargs will have devices? e.g. when the user doesnt explicitly specify devices for the input

Reply:
It is possible that the model has no parameters and there are no inputs to the forward method:

```py
def forward(seld):
    return torch.tensor([1, 2])
```

This is kind of a weird model. But it is possible, In any case, if the ```self._device``` is ```None```, we throw a ```RuntimeError```. So, we should be covered in this scenario.

If device is not specified, torch assumed device as ```cpu```:
```sh
>>> import torch
>>> x = torch.tensor([1, 2])
>>> x.device
device(type='cpu')
```